File: library/src/main/java/io/github/anderscheow/validator/rules/common/NotEmptyRule.java
package io.github.anderscheow.validator.rules.common;

import android.support.annotation.NonNull;
import android.support.annotation.StringRes;

import io.github.anderscheow.validator.rules.BaseRule;

public class NotEmptyRule extends BaseRule {

    public NotEmptyRule() {
        super("Value must not be empty");
    }

    public NotEmptyRule(@StringRes int errorRes) {
        super(errorRes);
    }

    public NotEmptyRule(@NonNull String errorMessage) {
        super(errorMessage);
    }

    @Override
    public boolean validate(Object value) {
        if (value == null) {
            throw new NullPointerException();
        }

        if (value instanceof String) {
            return !((String) value).isEmpty();
        }

        throw new ClassCastException("Required String value");
    }
}

File: library/src/main/java/io/github/anderscheow/validator/rules/common/NotEmptyRule.kt
package io.github.anderscheow.validator.rules.common

import androidx.annotation.StringRes
import io.github.anderscheow.validator.rules.Rule

class NotEmptyRule : Rule {

    constructor(@StringRes errorRes: Int) : super(errorRes)

    constructor(errorMessage: String) : super(errorMessage)

    override fun validate(value: String?): Boolean {
        if (value == null) {
            throw NullPointerException()
        } else {
            return value.isNotEmpty()
        }
    }
}

fun notEmpty(@StringRes errorRes: Int): NotEmptyRule = NotEmptyRule(errorRes)

fun notEmpty(errorMessage: String): NotEmptyRule = NotEmptyRule(errorMessage)
